prow/pod-utils/clone/clone.go

```diff
@@ -110,7 +110,12 @@ func PathForRefs(baseDir string, refs prowapi.Refs) string {
 	if refs.PathAlias != "" {
 		clonePath = refs.PathAlias
 	} else {
-		clonePath = fmt.Sprintf("github.com/%s/%s", refs.Org, refs.Repo)
+		repoUrl, err := url.Parse(refs.RepoLink)
```

Comment:
cc @sbueringer 

I don't think we should ever do an `iff err != nil { assume user intended to use GHE }`.

Also, please add some testcases demonstrating what you want to achieve.

Reply:
Can we confidently assume repoLink is a valid url? If so, we don't have to handle err. I will add test case as well. Thanks

Reply:
My point being: Don't make assumptions based on `if err != nil`. If you want to make assumptions, test for those.

As for the concrete change, its not clear to me what this does that you can't do with `refs.PathAlias`

Reply:
I agree we can achieve the same with ```refs.PathAlias``` but that would require all presubmit jobs to have ```refs.PathAlias``` and ```refs.CloneURI``` set other than ```github.com```. I'm wondering if we can achieve using the ```github.link_url``` or any other field

Reply:
I could also add it here [pjutil](https://github.com/kubernetes/test-infra/blob/master/prow/pjutil/pjutil.go#L72) where repoLink can be parsed. If it's done here no assumptions are needed as it's a required field. Thoughts?

Reply:
Would it make sense the other way round to default `PathAlias` based on the `CloneURI` (which would require a parsing that works with all possible git url types)?

Reply:
CloneURI is also optional field and is not set [here](https://github.com/kubernetes/test-infra/blob/master/prow/pjutil/pjutil.go#L72).  Would it be more appropriate to set CloneURI and PathAlias when new NewPresubmit is triggered in ```pjutil```? Like this.

```
func createRefs(pr github.PullRequest, baseSHA string) prowapi.Refs {
...
repoLink := pr.Base.Repo.HTMLURL
...
repoUrl, _ := url.Parse(repoLink)
return prowapi.Refs{
  ...
  CloneURI:  fmt.Sprintf("%s.git", repoLink),
  PathAlias: fmt.Sprintf("%s/%s/%s", repoUrl.Host, org, repo),
  ...
}

This would guarantee both cloneURI and pathAlias are always set. Let me know.